### ISSUE TITLE
Reproducing your results on ForensicHub Benchmark (Doc Protocol)

### ISSUE THREAD

**Issue by Mohamed-Dhouib** [KahimWong/ADCD-Net#13]
Hello,
I'm struggling to reproduce the results you shared for ForensicHub Benchmark (Doc Protocol).
Can you share a guide on how to proceed to reproduce them? Thanks!

**Comment by KahimWong** [comment]
Hi, thank you for the interest in ADCD-Net, to reprodoce the Doc Protocol results, please take the steps:

---

## 0) Get the code + environment ready + download data

1. Clone ADCD-Net and install dependencies.
2. Match the repo’s suggested environment: **Python 3.10.13**, **PyTorch 2.3.0 + CUDA 12.1**, **albumentations 2.0.8**. 

---

## 1) Download the required test data

### A) DocTamper (for Test / FCD / SCD)

* Download **DocTamper** data from https://github.com/qcf-568/DocTamper (you’ll need the official files including `qt_table.pk` and the `pks` contents) and corresponding OCR masks from https://drive.google.com/file/d/1-5BU3Bavs6SGJpaByua_FhDuUJGoo-iS/view?usp=sharing.

### B) The 4 additional Doc Protocol cross-domain sets (for T-SROIE / OSTF / TPIC-13 / RTM)

* Download the packaged sets from https://drive.google.com/file/d/18SFU1BHoBTcvhNnDLoDRqGdpDfzLvnJp/view?usp=sharing (we wrap up the 4 cross-domain test sets + corresponding OCR masks + path pickle files). 

---

## 2) Download model checkpoints 

From https://drive.google.com/file/d/1-5BU3Bavs6SGJpaByua_FhDuUJGoo-iS/view?usp=sharing, download:

* `ADCDNet.pth` (ADCD-Net checkpoint)
* `docres.pkl` (DocRes checkpoint)

---

## 3) Set evaluation config in `cfg.py`

In `cfg.py`, set:

* **GPUs** you want to use (or set `CUDA_VISIBLE_DEVICES`)
* `mode = 'val'`
* `all_ds_name = ['TestingSet', 'FCD', 'SCD', 'T-SROIE', 'TPIC-13', 'OSTF', 'RTM']`
  (Use the exact dataset names the loader expects—don’t append extra suffixes.)
* `root = '/path/to/DocTamper_root'`  (your DocTamper dataset root)
* `ckpt = '/path/to/ADCDNet.pth'`
* `docres_ckpt_path = '/path/to/docres.pkl'`
* `pkl_dir = '/path/to/path_pickles'` (where those `(img_path, mask_path, ocr_mask_path)` pickle files live)

For **FCD/SCD/Test**, samples are JPEG-compressed once using the final quality factor recorded in the official DocTamper pickle file; authentic images are skipped because there are no true positives .

---

## 4) Run evaluation

```bash
python main.py
```
---

**Comment by KahimWong** [comment]
Sure, we will upload the repo as soon as possbile.

**Comment by KahimWong** [comment]
Hi, please check our latest repo

**Comment by KahimWong** [comment]
We have updated our ADCD-Net repo

**Comment by Mohamed-Dhouib** [comment]
> > Sure
> 
> Hello again, Something I don't really understand: if training batch size/number of iters is still the same, why results have changed on doctamper in-domain data? (where evaluation was already correct as they are already 512*512 crops)

Also, your metric is still not the same as DocProtocol: You average precision and recall across images and then use them to calculate the F1 score. In DocProtocol, they compute the F1 score per image and then average the F1 scores.

Another thing is that you evaluate your model during training on part of the test sets, which normally isn't allowed (especially for cross-domain validation), so you should choose datasets that have validation sets to validate on, or just take the checkpoint at the end of training.

**Comment by KahimWong** [comment]
Please check our corrected script and results
